rfd/xxxx-bot-for-cert-renewals.md

```diff
@@ -0,0 +1,298 @@
+specific CA. We aim to simplify this configuration process for common systems by
+providing a `tbot config` command which spits out snippets that can be pasted 
+directly into a configuration file.
+
+For example, if `tbot` is running on an OpenSSH server and managing/renewing a
+host certificate, `tbot config` would render the correct `/etc/ssh/ssh_config`
+configuration for presenting the host certificate and trusting users signed by
+Teleport's user CA.
+
+Initially, `tbot config` will be limited to supporting OpenSSH configuration
+(client and server).
+
+##### Join Script
+
+We intend to develop a bot join script, similar in functionailty to the node
+join script that exists in Teleport enterprise. While not required, this will
+make it even easier to install and run the bot as a systemd service on the
+target machine.
+
+### Implementation
+
+Initially, the bot will be implemented as a standalone tool (ie under
+`tool/tbot`) and _not_ as a service that runs in a `teleport` process.
+
+The bulk of the new logic will be implemented in a new `renew` package, which
+will enable us to start with a standalone bot, and incorporate this
+functionality into other parts of Teleport (ie database access) in the future.
+
+### tbot configuration
+
+The goal is to keep tbot configuration to a minimum to make for a quick getting
+started experience. Ideally, all configuration is provided via the command line
+and there is no tbot-specific configuration file. This has several advantages:
+
+- Our users are less likely to mix up a tbot configuration file and a
+  teleport.yaml.
+- The command used to invoke tbot is all that is needed to run the same
+  configuration. This makes it easy to troubleshoot or recreate a customer
+  issue.
+
+#### Configuration Assist
+
+- `tbot config` to call a simple REST API that listens on loopback interface
+  only (it must be run on the same host as the tbot agent)
+- if tbot is managing multiple `--cert`s, you can pass `mode:destination` to
+  `tbot config` to tell it which cert to render configuration for
+
+#### Polling for expiration
+
+In order to watch a certificate for expiration, the
+`teleport/lib/utils/interval` package will be used. Note that tbot will only set
+up polling for _user certs_, as host certificates issued by teleport do not
+expire (host certificates *will* be re-issued during CA rotations).
+
+#### Renewable Certificates
+
+Teleport's current behavior when re-issuing a user certificate is maintain the
+original expiration and refuse to bump out the TTL. We plan to preserve this
+behavior by default, and only allow for extending the expiration date on
+certificates that have been marked renewable.
+
+In order to mark a certificate as renewable, we'll add an attribute to the
+subject in the certificate when the `bot` system role is present in the
+certificate identity.
+
+#### RBAC
+
+The new system role for the bot will need: (verify this)
```

Comment:
See the impersonation point above. It seems like there is no need for a hardcoded bot role, but instead each bot will have it's own unique impersonator role limiting it's scope.